### PR TITLE
Schema.org markup for datasets

### LINKED ISSUE #796: Add schema.org markup for datasets
Add schema.org markup to support datasets being picked up by Google Search for its Dataset Search. This should only be added to purl pages where the item is a dataset. @arcadiafalcone is mapping Public XML to Google's set of recommended schema.org fields. 

An object is a dataset if has a `<genre>` tag with the value `dataset`.

More info:

- Analysis on schema.org for Datasets with links to Google's documentation: https://docs.google.com/document/d/1BO10k_zSTqqT1YmlCg5oE4tOsGXiqmHzQyb6itZypwo/edit
- [SearchWorks implementatio](https://github.com/sul-dlss/SearchWorks/blob/f2ed40371750c9d31bfba0b2f48f4dbd9d76de79/app/models/solr_document.rb#L36)n (note that it does not include all required fields). 
- [Example schema.org markup](https://developers.google.com/search/docs/appearance/structured-data/dataset#example)  from Google

### PR DESCRIPTION
**Pull request by lwrubel**
Resolves #796 to create schema.org markup on PURL pages for datasets. 

Example schema.org JSON, produced for https://purl.stanford.edu/bb708mf8734:

`{"@context":"http://schema.org","@type":"Dataset","name":"Data on Quantifying the Viral Reduction Achieved Using Ash and Sand as Handwashing Agents","identifier":["https://doi.org/10.25740/bb708mf8734"],"description":"Volunteer experiment results described in the corresponding manuscript. ","isAccessibleForFree":true,"license":"https://creativecommons.org/licenses/by/4.0/legalcode","url":"https://purl.stanford.edu/bb708mf8734","creator":[{"@type":"Person","name":"Winnie Zambrana","givenName":"Winnie","familyName":"Zambrana"},{"@type":"Person","name":"Marlene Wolfe","givenName":"Marlene","familyName":"Wolfe"},{"@type":"Person","name":"Alexandria Boehm","givenName":"Alexandria","familyName":"Boehm"},{"@type":"Person","name":"Jingyan  Tong","givenName":"Jingyan ","familyName":"Tong"},{"@type":"Person","name":"Claire  Anderson","givenName":"Claire ","familyName":"Anderson"}]}`

and with many authors, one of which has an ORCID: (https://purl.stanford.edu/rn970zy4428):

`{"@context":"http://schema.org","@type":"Dataset","name":"16S rRNA data and metadata for \"Chemoproteomic identification of a DPP4 homolog in Bacteroides thetaiotaomicron\"","identifier":["https://doi.org/10.25740/rn970zy4428"],"description":"Serine hydrolases play important roles in signaling and human metabolism, yet little is known about their functions in gut commensal bacteria. Using bioinformatics and chemoproteomics, we identify serine hydrolases in the gut commensal Bacteroides thetaiotaomicron that are specific to the Bacteroidetes phylum. Two are predicted homologs of the human protease dipeptidyl peptidase 4 (hDPP4), a key enzyme that regulates insulin signaling. Functional studies reveal that BT4193 is a true homolog of hDPP4 that can be inhibited by FDA-approved type 2 diabetes medications targeting hDPP4, while the other is a misannotated proline-specific triaminopeptidase. We demonstrate that BT4193 is important for envelope integrity and that loss of BT4193 reduces B. thetaiotaomicron fitness during in vitro growth within a diverse community. However, neither function is dependent on BT4193 proteolytic activity, suggesting a scaffolding or signaling function for this bacterial protease.","isAccessibleForFree":true,"license":"https://opendatacommons.org/licenses/by/1-0/","url":"https://purl.stanford.edu/rn970zy4428","creator":[{"@type":"Person","name":"Laura J. Keller","givenName":"Laura J.","familyName":"Keller"},{"@type":"Person","name":"Taylor H. Nguyen","givenName":"Taylor H.","familyName":"Nguyen","sameAs":"https://orcid.org/0000-0002-1288-8882"},{"@type":"Person","name":"Lawrence Liu","givenName":"Lawrence","familyName":"Liu"},{"@type":"Person","name":"Brianna M.  Hurysz","givenName":"Brianna M. ","familyName":"Hurysz"},{"@type":"Person","name":"Markus Lakemeyer","givenName":"Markus","familyName":"Lakemeyer"},{"@type":"Person","name":"Matteo Guerra","givenName":"Matteo","familyName":"Guerra"},{"@type":"Person","name":"Danielle J. Gelsinger","givenName":"Danielle J.","familyName":"Gelsinger"},{"@type":"Person","name":"Rachael Chanin","givenName":"Rachael","familyName":"Chanin"},{"@type":"Person","name":"Nhi Ngo","givenName":"Nhi","familyName":"Ngo"},{"@type":"Person","name":"Kenneth M. Lum","givenName":"Kenneth M.","familyName":"Lum"},{"@type":"Person","name":"Franco Faucher","givenName":"Franco","familyName":"Faucher"},{"@type":"Person","name":"Phillip Ipock","givenName":"Phillip","familyName":"Ipock"},{"@type":"Person","name":"Micah J. Niphakis","givenName":"Micah J.","familyName":"Niphakis"},{"@type":"Person","name":"Ami S. Bhatt","givenName":"Ami S.","familyName":"Bhatt"},{"@type":"Person","name":"Anthony J. O'Donoghue","givenName":"Anthony J.","familyName":"O'Donoghue"},{"@type":"Person","name":"Kerwyn Casey Huang","givenName":"Kerwyn Casey","familyName":"Huang"},{"@type":"Person","name":"Matthew Bogyo","givenName":"Matthew","familyName":"Bogyo"}]}`

### CODE REVIEW

**Review by ndushay — COMMENTED** [review]
looks good.  Of course, I'm immediately thinking "um ...so how complex is our actual data for the fields indicated?   Ed's visualization might help with that question, but only kinda.

File: app/models/purl_resource.rb [hunk 1]
```diff
@@ -144,6 +144,10 @@ def collection_items_link
     "#{Settings.searchworks.url}/catalog?f[collection][]=#{folio_instance_hrid || druid}"
   end
 
+  def schema_dot_org?
```
Comment (justinlittman) [inline app/models/purl_resource.rb:147]: Can you add a comment to indicate what this means?

File: lib/metadata/schema_dot_org.rb [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      return true if dataset.any?
+
+      false
+    end
+
+    def title_name
+      # title.value or concatenated title.structuredValue 1) for title with status "primary" if present 2) for first title
+      # required for Datasets
+      titles = JsonPath.new("$.description.title[?(@['status' == 'primary'])].structuredValue[*].value").on(@cocina_json)
+      return titles.join('\n') unless titles.empty?
```
Comment (justinlittman) [inline lib/metadata/schema_dot_org.rb:51]: The concatenation with `\n` seems surprising.

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comments.

**Comment by lwrubel** [comment]
@ndushay just checking if there is anything further I should do for this PR? 

I checked with @arcadiafalcone as well to see if the examples looked correct and whether there were additional scenarios to test (let me know if you have further suggestions on that, @arcadiafalcone).

**Comment by ndushay** [comment]
I guess nothing actionable;  we'll find out (or not) if what we expose for schema.org is yucky - it's not realistic to expect the tests to show all the yuckies at this point.

**Pull request merged**